csrc/scheduler/normalization_inner_outer.cpp

```diff
@@ -317,6 +317,42 @@ struct PersistentBufferStorageParams {
   bool project_to_input = false;
 };
 
+// Prioritize moving buffers used by outer broadcast tensors to shared memory
+// because:
+// (1) They are reused in every iteration of the outer loop, has lower IO.
+// (2) Load occurs before the outer loop. Temporary register usage won't
+//     increase register pressure since the loop is the high-pressure region.
```

Comment:
> since the loop is the high-pressure region.

Just to confirm that I read this correctly: bcast_buffer is loaded before the outer loop, which impacts register usage and that's why we are moving it to shared memory.

Reply:
Note to myself. offline discussion answered that my understanding is correct. I'll take another look at the code comment.